src/Middleware/RequestDecompression/src/DeflateDecompressionProvider.cs

```diff
@@ -0,0 +1,21 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO.Compression;
+
+namespace Microsoft.AspNetCore.RequestDecompression;
+
+/// <summary>
+/// DEFLATE decompression provider.
+/// </summary>
+public class DeflateDecompressionProvider : IDecompressionProvider
+{
+    /// <inheritdoc />
+    public string EncodingName => "deflate";
+
+    /// <inheritdoc />
+    public Stream CreateStream(Stream outputStream)
+    {
+        return new DeflateStream(outputStream, CompressionMode.Decompress);
```

Comment:
Aren't these vulnerable to zip bombs?

Reply:
That's a good point. I suppose the RequestDecompressionProvider could keep track of the number of bytes read from the stream and then stop/throw if this passes some threshold. Although I'm not super familiar with the details of decompression bombs, so there might be more that I'm missing. I'll look into this further.

Reply:
It's very likely we would want to have some sort of maximum size limits on these. Perhaps require putting some metadata on endpoints to indicate that the request should be decompressed with a max-allowed size?

Spit balling

```C#
[AllowRequestDecompression(maxDecompressedSize: /* size in bytes */)]
public IResult Post(Stream requestStream) => ...
```

Reply:
We should re-use the normal request size limits and metadata.
https://github.com/dotnet/aspnetcore/blob/main/src/Http/Http.Features/src/IHttpMaxRequestBodySizeFeature.cs

Reply:
Could this be a middleware global option, that can also be set using the attribute? With a sensible default in the main option.

Reply:
I added a `MaxRequestBodySize` option to `RequestDecompressionOptions` and a `SetRequestSizeLimit` method to the `RequestDecompressionProvider` to set the limit the same way as `RequestSizeLimitAttribute`. Still need to add a default value though.

Reply:
I don't think you should need MaxRequestBodySize on the options, you should be able to take the current value from IHttpMaxRequestBodySizeFeature.MaxRequestBodySize which will be set by the servers. You'd update this with RequestSizeLimitAttribute data from the current endpoint Endpoint if any.

Reply:
Sounds good, I'll remove this from `RequestDecompressionOptions`.